### PR TITLE
Tweak pseudo-builder in user guide

### PR DESCRIPTION
**Pull request by mwichmann**
Minor fiddling - example format, headers, wording tweaks.

Doc-only change - this is minor enough it doesn't even seem worth mentioning in RELEASE.

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `CHANGES.txt` (and read the `README.rst`)
* [X] I have updated the appropriate documentation

### CODE REVIEW

**Comment by bdbaddog** [comment]
Weird. I can't get this to rebuild 

```
Running Sphinx v4.5.0
loading translations [en]... done

Extension error:
Could not import extension rst2pdf.pdfbuilder (exception: dlopen(/Users/bdbaddog/.virtualenvs/scons37/lib/python3.7/site-packages/PIL/_imaging.cpython-37m-darwin.so, 0x0002): Library not loaded: '/opt/local/lib/libtiff.5.dylib'
  Referenced from: '/Users/bdbaddog/.virtualenvs/scons37/lib/python3.7/site-packages/PIL/_imaging.cpython-37m-darwin.so'
  Reason: tried: '/opt/local/lib/libtiff.5.dylib' (no such file), '/usr/local/lib/libtiff.5.dylib' (no such file), '/usr/lib/libtiff.5.dylib' (no such file))
make: *** [pdf] Error 2
scons: *** [build/doc/api/scons-api.pdf] Error 2
scons: building terminated because of errors.
```

**Comment by mwichmann** [comment]
Heh: snarky comment: don't use a five-year-old Python :)    Sphinx comes with problems, but shouldn't be triggerable by a bit of xml in the user guide (which, in fact, should have built by the time you hit that problem, whatever it is)

File: doc/user/add-method.xml [hunk 1]
```diff
@@ -89,31 +70,35 @@ int main() { printf("Hello, world!\n"); }
 
   <para>
 
-  As mentioned, a pseudo-builder also provides more flexibility
-  in parsing arguments than you can get with a &Builder;.
+  A pseudo-builder is useful because it provides more flexibility
+  in parsing arguments than you can get with a standard Builder method.
```
Comment (bdbaddog) [inline doc/user/add-method.xml]: Curious why de-entity Builder?
Also Builders' aren't always methods, for example in the DefaultEnvironment().. they are accessible  without the env.

Comment (mwichmann) [reply doc/user/add-method.xml]: It's too awkward to keep going method-or-function all the time. Can of course do so.

Meanwhile, builders are things, didn't seem to me like they need special markup. Also not a big deal.

Comment (bdbaddog) [reply doc/user/add-method.xml]: I think it's valuable to markup builder as it's not the verb builder but rather the abstraction builder which  part of SCons.

**Comment by mwichmann** [comment]
So I admit this is nitpicking, but the markup for builder is as follows:
```xml
<!ENTITY Builder "<classname xmlns='http://www.scons.org/dbxsd/v1.0'>Builder</classname>">
```

It's not a class... really.  `Builder()` itself is a factory function and appears as an environment method, but we were talking about _builder-the-concept_ rather than _Builder-the-OOP-thing_.  And for the OOP thing I can use `&f-link-Builder;`. Is there a better way to mark this up? The same is true of Scanner, of course.

Docbook doesn't really have markup for concepts, except as terms in a definition list (`<term>` can't be used standalone).

**Comment by bdbaddog** [comment]
IMHO, while it's not perfect, the current formatting indicates to the reader that it's not just builder the verb which is being discussed, but builder as it relates to SCons, which I think is sufficient, since we don't have two distinct terms to differentiate (at least not two terms which wouldn't be painful to use again and again)

**Comment by mwichmann** [comment]
Okay, proposal.  My complaint about markup actually applies to many things in `scons.mod`, but I'm not going there.

How about `<glossterm>term</glossterm>` for concepts (I'll only change Builder and Scanner at this time if so)?  These are things that we might actually have definitions of in a Glossary someday, if only we were producing a glossary, and at that point they would turn up as links to the glossary entry; for now it just renders in italics (rather than "code").

**Comment by bdbaddog** [comment]
I like that. Will the formatting of the produced text be different from the entity we have now?

**Comment by mwichmann** [comment]
> I like that. Will the formatting of the produced text be different from the entity we have now?

yes. Italics now.

File: doc/user/scanners.xml [hunk 2]
```diff
@@ -146,15 +146,16 @@ over the file scanning rather than being called for each input line:
 
   <para>
 
-    &SCons; has built-in scanners that know how to look in
+    &SCons; has routines that know how to look in
```
Comment (bdbaddog) [inline doc/user/scanners.xml]: I'd leave scanner.

**Comment by mwichmann** [comment]
Wrong button, sorry

File: doc/user/scanners.xml (hunk 2 above)
Comment (mwichmann) [reply doc/user/scanners.xml]: This was an attempt to step around the plural problem without having gone to the trouble of actually addressing it, which I'll get to someday.

Comment (mwichmann) [reply doc/user/scanners.xml]: Hint: don't bother using the github phone app to ever *do* anything with an issue, it's just too limited, and the opposite of straightforward.

Comment (mwichmann) [reply doc/user/scanners.xml]: This has been updated.

File: doc/user/scanners.xml [hunk 3]
```diff
@@ -146,15 +146,15 @@ over the file scanning rather than being called for each input line:
 
   <para>
 
-    &SCons; has built-in scanners that know how to look in
+    &SCons; has built-in &Scanners; that know how to look in
     C/C++, Fortran, D, IDL, LaTeX, Python and SWIG source files
     for information about
-    other files that targets built from those files depend on--for example,
-    in the case of files that use the C preprocessor,
+    other files that targets built from those files depend on -
```
Comment (bdbaddog) [inline doc/user/scanners.xml]: Is there meant to be a trailing `-` ?

**Pull request merged**